x-pack/plugins/security_solution/public/management/pages/policy/store/policy_details/middleware/policy_trusted_apps_middleware.ts

```diff
@@ -5,12 +5,192 @@
  * 2.0.
  */
 
-import { MiddlewareRunner } from '../../../types';
+import { find, isEmpty } from 'lodash/fp';
+import { PolicyDetailsState, MiddlewareRunner } from '../../../types';
+import {
+  policyIdFromParams,
+  isOnPolicyTrustedAppsPage,
+  getCurrentArtifactsLocation,
+  getAvailableArtifactsList,
+} from '../selectors';
+import {
+  ImmutableArray,
+  PostTrustedAppCreateRequest,
+} from '../../../../../../../common/endpoint/types';
+import { ImmutableMiddlewareAPI } from '../../../../../../common/store';
+import { TrustedAppsHttpService, TrustedAppsService } from '../../../../trusted_apps/service';
+import {
+  createLoadedResourceState,
+  createLoadingResourceState,
+  createUninitialisedResourceState,
+  createFailedResourceState,
+} from '../../../../../state';
+import { parseQueryFilterToKQL } from '../../../../../common/utils';
+import { SEARCHABLE_FIELDS } from '../../../../trusted_apps/constants';
+import { PolicyDetailsAction } from '../action';
+
+const checkIfThereAreAvailableTrustedApps = async (
+  store: ImmutableMiddlewareAPI<PolicyDetailsState, PolicyDetailsAction>,
+  trustedAppsService: TrustedAppsService
+) => {
+  const state = store.getState();
+  const policyId = policyIdFromParams(state);
+
+  store.dispatch({
+    type: 'policyArtifactsAvailableListExistDataChanged',
+    // Ignore will be fixed with when AsyncResourceState is refactored (#830)
+    // @ts-ignore
+    payload: createLoadingResourceState({ previousState: createUninitialisedResourceState() }),
+  });
+  try {
+    const trustedApps = await trustedAppsService.getTrustedAppsList({
+      page: 1,
+      per_page: 100,
+      kuery: `(not exception-list-agnostic.attributes.tags:"policy:${policyId}") AND (not exception-list-agnostic.attributes.tags:"policy:all")`,
```

Comment:
is `kuery` really correct? should it be `query`?

Reply:
Yeah, it's `kuery` because it's a kql query. So the API field is `kuery`

Reply:
Oh wow! that's funny. Thanks for the clarification :D